torch/utils/_mode_utils.py

```diff
@@ -97,11 +97,15 @@ def _enable_mode(mode, mode_info: _ModeInfo, *, replace=None, ignore_preexisting
         raise ValueError(
             f'The argument passed to enable_{mode_info.mode_name}_mode must implement {required_fn}'
         )
+    if hasattr(mode, "setup_mode"):
+        mode.setup_mode()  # type: ignore[attr-defined]
     mode_info.set_mode(mode)
     try:
         yield
     finally:
         mode_info.set_mode(old)
+        if hasattr(mode, "cleanup_mode"):
+            mode.cleanup_mode()  # type: ignore[attr-defined]
```

Comment:
We should be able to get rid of this under the new design right

Reply:
yup - working on cleaning up, had to debug issue with new-style modes. when i requested another review didn't realize there was workaround to bug I hit